.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+  test_matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
+        ruby: [2.7, '3.0']
+        # Test against multiple Rails versions
+        gemfile: [rails_6, rails_7]
+    runs-on: ubuntu-latest
+    env:
+      DB_DATABASE: signonotron2_test
+      DB_USER: root
+      DB_PASSWORD: root
+      MYSQL_ALLOW_EMPTY_PASSWORD: true
+      BUNDLE_GEMFILE: gemfiles/${{ matrix.gemfile }}.gemfile
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7 # for signon
+          bundler-cache: true
+          if: ${{ matrix.ruby != 2.7 }}
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Set up MySQL
+        run: |
+          sudo /etc/init.d/mysql start
+          mysql -e "CREATE DATABASE ${{ env.DB_DATABASE }}; CREATE USER 'signonotron2'@'localhost' IDENTIFIED BY 'signonotron2'; GRANT ALL ON ${{ env.DB_DATABASE }}.* to 'signonotron2'@'localhost';" -u${{ env.DB_USER }} -p${{ env.DB_PASSWORD }}
+      - name: Start Signon
+        run: |
+          export PATH=/opt/hostedtoolcache/Ruby/2.7.6/x64/bin:$PATH
+          git clean -fxde /tmp
+          env BUNDLE_GEMFILE=Gemfile TEST_DATABASE_URL=mysql2://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:3306/${{ env.DB_DATABASE }} ./start_signon.sh
+      - name: Run tests
+        run: |
+          bundle install
+          bundle exec rake
+      - if: always()
+        name: Stop Signon
+        run: |
+          ./stop_signon.sh
+
+  # This job is needed to work around the fact that matrix jobs spawn multiple status checks – i.e. one job per variant.
+  # The branch protection rules depend on this as a composite job to ensure that all preceding test_matrix checks passed.
+  # Solution taken from: https://github.community/t/status-check-for-a-matrix-jobs/127354/3
+  test:
+    needs: test_matrix
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "All matrix tests have passed 🚀"
+
+  release:
+    needs: test
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/main' }}
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          rubygems: latest
+      - env:
+          GEM_HOST_API_KEY: ${{ secrets.ALPHAGOV_RUBYGEMS_API_KEY }}
+        run: |
+          VERSION=$(ruby -e "puts eval(File.read('gds-sso.gemspec')).version")
+          GEM_VERSION=$(gem list --exact --remote gds-sso)
+
+          if [ "${GEM_VERSION}" != "gds-sso (${VERSION})" ]; then
+            gem build gds-sso.gemspec
+            gem push "gds-sso-${VERSION}.gem"
+          fi
+
+          if ! git ls-remote --tags --exit-code origin v${VERSION}; then
+            git tag v${VERSION}
+            git push --tags
+          fi
```

Comment:
Probably it's the PR title that's wrong. This is actually left behind from my previous attempt to migrate this repo to GH actions, and I hadn't actually looked at this step this time around (I think it's copied from another gem).

The important part is certainly the testing, since it's blocking another PR.

But unless we have a pressing desire to keep the release phase on Jenkins I think we could just leave this in (and I'll rename the PR accordingly).